### ISSUE TITLE
Unicode error when piping to file

### ISSUE THREAD

**Issue by zmitchell** [jonathanj/eliottree#95]
Hi, I'm trying to pipe the output of `eliot-tree` to a file on Windows 10, but I'm getting the following traceback:
```
Traceback (most recent call last):
  File "c:\users\zach\appdata\local\programs\python\python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\zach\appdata\local\programs\python\python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\Zach\AppData\Local\pypoetry\Cache\virtualenvs\ns-trcd-5J_fInKt-py3.7\Scripts\eliot-tree.exe\__main__.py", line 9, in <module>
  File "c:\users\zach\appdata\local\pypoetry\cache\virtualenvs\ns-trcd-5j_finkt-py3.7\lib\site-packages\eliottree\_cli.py", line 318, in main
    theme_overrides=config.get('theme_overrides'))
  File "c:\users\zach\appdata\local\pypoetry\cache\virtualenvs\ns-trcd-5j_finkt-py3.7\lib\site-packages\eliottree\_cli.py", line 126, in display_tasks
    theme=theme)
  File "c:\users\zach\appdata\local\pypoetry\cache\virtualenvs\ns-trcd-5j_finkt-py3.7\lib\site-packages\eliottree\_render.py", line 288, in render_tasks
    write(format_tree(task, _format_node, _get_children, make_options()))
  File "c:\users\zach\appdata\local\programs\python\python37\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u21d2' in position 71: character maps to <undefined>
```

The file I'm using looks like this:
```
{"action_status": "started", "timestamp": 1582840009.482032, "task_uuid": "6ffa3c9b-3e24-4ee7-bb57-2e40e01171b6", "action_type": "_set_initial_widget_states", "task_level": [1]}
{"new_max": 5, "action_status": "started", "timestamp": 1582840009.482032, "task_uuid": "6ffa3c9b-3e24-4ee7-bb57-2e40e01171b6", "action_type": "update_max_measurements", "task_level": [2, 1]}
{"action_status": "succeeded", "timestamp": 1582840009.482032, "task_uuid": "6ffa3c9b-3e24-4ee7-bb57-2e40e01171b6", "action_type": "update_max_measurements", "task_level": [2, 2]}
{"action_status": "succeeded", "timestamp": 1582840009.482032, "task_uuid": "6ffa3c9b-3e24-4ee7-bb57-2e40e01171b6", "action_type": "_set_initial_widget_states", "task_level": [3]}
{"new_max": 50, "action_status": "started", "timestamp": 1582840013.952636, "task_uuid": "ab19b06e-fb4d-406f-be82-109b5530df3b", "action_type": "update_max_measurements", "task_level": [1]}
{"action_status": "succeeded", "timestamp": 1582840013.952636, "task_uuid": "ab19b06e-fb4d-406f-be82-109b5530df3b", "action_type": "update_max_measurements", "task_level": [2]}
{"action_status": "started", "timestamp": 1582840018.5424337, "task_uuid": "9a2da24e-bf98-442e-b158-7b84845f7a0b", "action_type": "set_time_axis", "task_level": [1]}
{"action_status": "succeeded", "timestamp": 1582840018.5424337, "task_uuid": "9a2da24e-bf98-442e-b158-7b84845f7a0b", "action_type": "set_time_axis", "task_level": [2]}
{"action_status": "started", "timestamp": 1582840023.4722626, "task_uuid": "6c6ecdac-b624-4e88-849a-91ffee883af5", "action_type": "close", "task_level": [1]}
{"action_status": "succeeded", "timestamp": 1582840023.4722626, "task_uuid": "6c6ecdac-b624-4e88-849a-91ffee883af5", "action_type": "close", "task_level": [2]}
```

I've tried making the encoding process as simple as possible by using the `--ascii` and `--color never` flags, but the issue persists. Am I missing something simple, or is this a bug?

**Comment by romanstrazanec** [comment]
I also get the similar error running `!eliot-tree out.log` in **jupyter notebook**.

```
Traceback (most recent call last):
  File "c:\program files\python38\lib\runpy.py", line 192, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\program files\python38\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\vza7982\AppData\Roaming\Python\Python38\Scripts\eliot-tree.exe\__main__.py", line 7, in <module>
  File "C:\Users\vza7982\AppData\Roaming\Python\Python38\site-packages\eliottree\_cli.py", line 308, in main
    display_tasks(
  File "C:\Users\vza7982\AppData\Roaming\Python\Python38\site-packages\eliottree\_cli.py", line 116, in display_tasks
    render_tasks(
  File "C:\Users\vza7982\AppData\Roaming\Python\Python38\site-packages\eliottree\_render.py", line 288, in render_tasks
    write(format_tree(task, _format_node, _get_children, make_options()))
  File "c:\program files\python38\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode characters in position 38-40: character maps to <undefined>
```

**Comment by jonathanj** [comment]
@romanstrazanec The minimum case that reproduces this would be useful.

**Comment by jonathanj** [comment]
@zmitchell I missed this issue last year when you posted it, sorry about that. If you’re still interested in helping diagnose this, could you provide some information about your operating system and locale? Your input doesn’t look like it contains any Unicode and the options you mentioned should disable all other Unicode output, but there may be some lingering Unicode. 

You could also try changing the code page of cmd to a UTF-8 capable one. 

cc @romanstrazanec

**Comment by jonathanj** [comment]
See if any of these suggestions help. https://stackoverflow.com/questions/492483/setting-the-correct-encoding-when-piping-stdout-in-python

**Comment by romanstrazanec** [comment]
@jonathanj I used the @zmitchell's log file, or it could have been with any file.

The problem was with:
⇒ Rightwards Double Arrow U+21D2 
⧖ Hourglass U+29D6

`--ascii` flag did not change those characters to ascii style.

I created pull request.

**Comment by jonathanj** [comment]
This fix is available in the 21.0.0 release.
